src/VisualStudio/Core/Def/ServiceHubDebuggingService.cs

```diff
@@ -0,0 +1,293 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Management;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using DTE = EnvDTE.DTE;
+
+namespace Microsoft.VisualStudio.LanguageServices;
+
+/// <summary>
+/// This class is a debugging aid to help attach the VS debugger to the Roslyn ServiceHub process.
```

Comment:
Is this a feature we should just check into the ServiceHub repository so everybody gets this? I'd imagine it also might make it easier to do because rather than having to do all the trickery of watching for the process to launch and figuring out if it's a the right child process, that code would just now since it launched the process in the first place?